test/e2e/framework/node/resource.go

```diff
@@ -67,51 +67,40 @@ func FirstAddress(nodelist *v1.NodeList, addrType v1.NodeAddressType) string {
 	// Check the node readiness condition (logging all).
+	// Ensure that the condition type and the status matches as desired.
 	for _, cond := range node.Status.Conditions {
-		// Ensure that the condition type and the status matches as desired.
-		if cond.Type == conditionType {
-			// For NodeReady condition we need to check Taints as well
-			if cond.Type == v1.NodeReady {
-				hasNodeControllerTaints := false
-				// For NodeReady we need to check if Taints are gone as well
-				taints := node.Spec.Taints
-				for _, taint := range taints {
-					if taint.MatchTaint(nodectlr.UnreachableTaintTemplate) || taint.MatchTaint(nodectlr.NotReadyTaintTemplate) {
-						hasNodeControllerTaints = true
-						break
-					}
-				}
-				if wantTrue {
-					if (cond.Status == v1.ConditionTrue) && !hasNodeControllerTaints {
-						return true
-					}
-					msg := ""
-					if !hasNodeControllerTaints {
-						msg = fmt.Sprintf("Condition %s of node %s is %v instead of %t. Reason: %v, message: %v",
-							conditionType, node.Name, cond.Status == v1.ConditionTrue, wantTrue, cond.Reason, cond.Message)
-					} else {
-						msg = fmt.Sprintf("Condition %s of node %s is %v, but Node is tainted by NodeController with %v. Failure",
-							conditionType, node.Name, cond.Status == v1.ConditionTrue, taints)
-					}
-					if !silent {
-						e2elog.Logf(msg)
-					}
-					return false
+		// For NodeReady condition we need to check Taints as well
+		if cond.Type == v1.NodeReady {
+			hasNodeControllerTaints := false
+			// For NodeReady we need to check if Taints are gone as well
+			taints := node.Spec.Taints
+			for _, taint := range taints {
+				if taint.MatchTaint(nodectlr.UnreachableTaintTemplate) || taint.MatchTaint(nodectlr.NotReadyTaintTemplate) {
+					hasNodeControllerTaints = true
+					break
 				}
-				// TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default
-				if cond.Status != v1.ConditionTrue {
+			}
+			if wantTrue {
+				if (cond.Status == v1.ConditionTrue) && !hasNodeControllerTaints {
 					return true
 				}
-				if !silent {
-					e2elog.Logf("Condition %s of node %s is %v instead of %t. Reason: %v, message: %v",
+				msg := ""
+				if !hasNodeControllerTaints {
+					msg = fmt.Sprintf("Condition %s of node %s is %v instead of %t. Reason: %v, message: %v",
 						conditionType, node.Name, cond.Status == v1.ConditionTrue, wantTrue, cond.Reason, cond.Message)
+				} else {
+					msg = fmt.Sprintf("Condition %s of node %s is %v, but Node is tainted by NodeController with %v. Failure",
+						conditionType, node.Name, cond.Status == v1.ConditionTrue, taints)
+				}
+				if !silent {
+					e2elog.Logf(msg)
 				}
 				return false
 			}
-			if (wantTrue && (cond.Status == v1.ConditionTrue)) || (!wantTrue && (cond.Status != v1.ConditionTrue)) {
+			// TODO: check if the Node is tainted once we enable NC notReady/unreachable taints by default
```

Comment:
this TODO is for sig-node to resolve if they want to.
taint based eviction is enabled by default in 1.13:
https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#taint-based-evictions

Reply:
yes,i want to resolve this if i can in later.